src/scripts/components/lang-redirects.js

```diff
@@ -0,0 +1,100 @@
+import Cookies from 'js-cookie';
+
+const allowedLanguages = ['en', 'ja', 'fr'];
+const enabledSubdomains = [
+	'docs',
+	'docs-staging',
+	'corpsite-preview',
+	'corpsite-staging',
+	'localhost'
+];
+const cookiePath = '/';
+
+const getUrlVars = () => {
+	const vars = {};
+	const href = window.location.href.replace(window.location.hash, '');
+    href.replace(/[?&]+([^=&]+)=([^&]*)/gi, function(m,key,value) {
+        vars[key] = value;
+    });
+    return vars;
+}
+
+const getDomain = (data) => {
+    const a = document.createElement('a');
+    a.href = data;
+    return a.hostname;
+}
+
+const getPath = (data) => {
+    const a = document.createElement('a');
+    a.href = data;
+    return a.pathname;
+}
+
+export function handleLanguageBasedRedirects() {
+	const params = getUrlVars();
+
+	const supportedLanguages = navigator.language || navigator.browserLanguage;
```

Comment:
running this gave me `en-US` would it give us `ja-JP` or anything for japan or is it just ja?

Reply:
Yes should be ja-JP for Japan https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLanguage/languages